commerce-api/src/main/java/com/liferay/commerce/search/facet/SerializableMultiValueFacet.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.search.facet;
+
+import com.liferay.portal.kernel.search.BooleanClause;
+import com.liferay.portal.kernel.search.BooleanClauseFactoryUtil;
+import com.liferay.portal.kernel.search.BooleanClauseOccur;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.facet.MultiValueFacet;
+import com.liferay.portal.kernel.search.facet.util.FacetValueValidator;
+import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.search.filter.TermsFilter;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.io.Serializable;
+
+/**
+ * @author Andrea Di Giorgi
+ */
+public class SerializableMultiValueFacet extends MultiValueFacet {
+
+	public SerializableMultiValueFacet(SearchContext searchContext) {
+		super(searchContext);
+	}
+
+	public boolean isNegated() {
+		return _negated;
+	}
+
+	public void setNegated(boolean negated) {
+		_negated = negated;
+	}
+
+	@Override
+	protected BooleanClause<Filter> doGetFacetFilterBooleanClause() {
+		SearchContext searchContext = getSearchContext();
+
+		Serializable serializableValues = searchContext.getAttribute(
+			getFieldName());
+
+		String[] values;
```

Comment:
initialize variable to `null`. And, actually, move this declaration above `Serializable serializableValues`

Reply:
Hmmm.... this section is actually a little weird. We call `searchContext.getAttribute(getFieldName())` twice, but we deal with the return value very differently each time. We should be more consistent here. Makes sense to me to do it like

    String[] values = GetterUtil.getStringValues(searchContext.getAttribute(getFieldName()));
    if (ArrayUtil.isEmpty(values)) {
        values = StringUtil.split(
				GetterUtil.getString(
					searchContext.getAttribute(getFieldName())));
    }

but if you don't like that, we should at least reuse `serializableValues` rather than re-calling `getAttribute`.

Reply:
Yes that works perfectly, so much better!